components/core/src/ffi/ir_stream/decoding_methods.cpp

```diff
@@ -0,0 +1,519 @@
+        static_assert(is_same_v<encoded_variable_t, eight_byte_encoded_variable_t> ||
+                      is_same_v<encoded_variable_t, four_byte_encoded_variable_t>);
+
+        if (tag == cProtocol::Payload::VarStrLenUByte ||
+            tag == cProtocol::Payload::VarStrLenUShort ||
+            tag == cProtocol::Payload::VarStrLenInt) {
+            is_encoded_var = false;
+            return true;
+        }
+
+        if constexpr (is_same_v<encoded_variable_t, eight_byte_encoded_variable_t>) {
+            if (tag == cProtocol::Payload::VarEightByteEncoding) {
+                is_encoded_var = true;
+                return true;
+            }
+        } else {
+            if (tag == cProtocol::Payload::VarFourByteEncoding) {
+                is_encoded_var = true;
+                return true;
+            }
+        }
+        return false;
+    }
+
+    template <typename integer_t>
+    static bool read_data_big_endian (IRBuffer& ir_buf, integer_t& data) {
+        constexpr auto read_size = sizeof(integer_t);
+        static_assert(read_size == 1 || read_size == 2 || read_size == 4 || read_size == 8);
+
+        integer_t value_small_endian;
+        if (ir_buf.try_read(value_small_endian) == false) {
+            return false;
+        }
+
+        if constexpr (read_size == 1) {
+            data = value_small_endian;
+        } else if constexpr (read_size == 2) {
+            data = bswap_16(value_small_endian);
+        } else if constexpr (read_size == 4) {
+            data = bswap_32(value_small_endian);
+        } else if constexpr (read_size == 8) {
+            data = bswap_64(value_small_endian);
+        }
+        return true;
+    }
+
+    static IRErrorCode
+    get_logtype_length (IRBuffer& ir_buf, encoded_tag_t encoded_tag, size_t& logtype_length) {
+        if (encoded_tag == cProtocol::Payload::LogtypeStrLenUByte) {
+            uint8_t length;
+            if (false == read_data_big_endian(ir_buf, length)) {
+                return IRErrorCode_InComplete_IR;
+            }
+            logtype_length = length;
+        } else if (encoded_tag == cProtocol::Payload::LogtypeStrLenUShort) {
+            uint16_t length;
+            if (false == read_data_big_endian(ir_buf, length)) {
+                return IRErrorCode_InComplete_IR;
+            }
+            logtype_length = length;
+        } else if (encoded_tag == cProtocol::Payload::LogtypeStrLenInt) {
+            int32_t length;
+            if (false == read_data_big_endian(ir_buf, length)) {
+                return IRErrorCode_InComplete_IR;
+            }
+            logtype_length = length;
+        } else {
+            return IRErrorCode_Corrupted_IR;
+        }
+        return IRErrorCode_Success;
+    }
+
+    static IRErrorCode
+    parse_logtype (IRBuffer& ir_buf, encoded_tag_t encoded_tag, string_view& logtype) {
+        size_t log_length;
+        if (IRErrorCode error_code = get_logtype_length(ir_buf, encoded_tag, log_length);
```

Comment:
Can we inline `get_logtype_length` to be consistent with `parse_dictionary_var`?